lesson14/index.js

```diff
@@ -0,0 +1,117 @@
+const ul = document.getElementById('js-list');
+const button = document.getElementById('js-button');
+const modal = document.getElementById('js-modal');
+const modalButton = document.getElementById('js-modal-button');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.id = "js-loading";
+    loadingImg.src = 'loading-circle.gif';
+    ul.appendChild(li).appendChild(loadingImg);
+};
+
+const showModal = () => {
+  modal.classList.add('isShow');
+}
+
+const hideModal = () => {
+  modal.classList.remove('isShow');
+}
+
+const removeLoading = () => document.getElementById('js-loading').remove();
+
+const removeButton = () => document.getElementById('js-button').remove();
+
+
+const createErrorMessage = (error) => {
+    const li = document.createElement("li");
+    li.textContent =`エラー内容：${error.message}`;
+    ul.appendChild(li);
+    console.error(error.message);
+};
+
+/**
+ * getInputNumber関数
+ * モーダル内のinputの値を取得し、consoleに出力する
+ */
+ const getInputNumber = () => {
+  const inputNumber = document.getElementById('js-input-number');
+  console.log(inputNumber.value);
+};
+
+/**
+ * Async関数 
+ * @return {Array} 
+ */
+const fetchData = async (endpoint) => {
+    try {
+        const response = await fetch(endpoint) // 中身はresponseオブジェクトを含むpromise
+        const json = await response.json(); //response オブジェクトから JSON を抽出
+        return json.data;
+
+    } catch(error) {
+        throw new Error(error);
+    } 
+};
+
+/**
+ * Async関数
+ * fetchData関数から値が返ってきた後、その値を返す
+ */
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/5tqv';
+const fetchListData = async () => {
+    createLoading();
+    getInputNumber(); // promiseが返ってくるfetchData関数の前でgetInputNumberを実行する
```

Comment:
IMO
I didn't think it was appropriate to run getInputNumber within fetchListData().
The reason is that it doesn't fit the function name.


I thought it would be a good idea to run it before createList()
```js
modalButton.addEventListener('click', () => {
  hideModal();
  getInputNumber();
  createList();
  removeButton();
});

Reply:
I agree that the function names do not match.
I have fixed the code.
[2012b71](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/14/commits/2012b71a0b8231344b71f6a5efd95df383011cb7)